ReactWindows/ReactNative.Shared/UIManager/AccessibleAutomationPeer.cs

```diff
@@ -0,0 +1,104 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using ReactNative.UIManager.Events;
+#if WINDOWS_UWP
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Automation.Provider;
+#else
+using System.Windows;
+using System.Windows.Automation.Peers;
+using System.Windows.Automation.Provider;
+#endif
+
+namespace ReactNative.UIManager
+{
+    /// <summary>
+    /// Automation peer for a FrameworkElement that implements IAccessible.
+    /// </summary>
+    /// <typeparam name="T">Type of owner of the automation peer.</typeparam>
+    public sealed class AccessibleAutomationPeer<T> : FrameworkElementAutomationPeer, IInvokeProvider
+        where T : FrameworkElement, IAccessible
+    {
+        /// <summary>
+        /// Hides base UIElement Owner to provide stronger-typed T Owner.
+        /// </summary>
+        private new T Owner => (T)base.Owner;
+
+        /// <inheritdoc />
+        public AccessibleAutomationPeer(T owner)
+            : base(owner)
+        {
+        }
+
+        /// <inheritdoc />
+        protected override IList<AutomationPeer> GetChildrenCore()
+        {
+            bool shouldHideChildren =
+                Owner.ImportantForAccessibility == ImportantForAccessibility.Yes
+                || Owner.ImportantForAccessibility == ImportantForAccessibility.NoHideDescendants;
+            return shouldHideChildren ? null : base.GetChildrenCore();
+        }
+
+        /// <inheritdoc />
+        protected override bool IsContentElementCore()
```

Comment:
>IsContentElementCore [](start = 32, length = 20)

I'd like to get a walk through of the meaning of these API's in person

Reply:
Sure. Ping me when you have time.